src/pages/create-account/choose-password.tsx

```diff
@@ -1,115 +1,94 @@
-			return
-		}
-
-		if (!canNext || !mnemonic || loading) {
-			return
-		}
+	const [password, setPassword] = useState<string>('')
+	const [password2, setPassword2] = useState<string>('')
 
+	const onClick = () => {
 		setLoading(true)
-		onNext(await wallet.encrypt(passwords[0]))
-		setLoading(false)
+		const wallet = Wallet.createRandom()
+		wallet.encrypt(password).then((encryptedWallet) => {
+			setProfile({
+				...profile,
+				encryptedWallet,
+				address: wallet.address,
+			})
+			setLoading(false)
+			navigate(ACCOUNT_CREATED)
+		})
 	}
 
-	return (
-		<div class="flex flex-col">
-			<h1 class="text-3xl mb-8">Choose a password.</h1>
-			{showPrompt ? (
-				<div class="w-full indicator mb-8 mt-6">
-					<div class="indicator-item indicator-top indicator-center badge h-12">
-						<ExclamationIcon class="w-8 h-8 " />
-					</div>
-					<div class="card shadow place-items-center">
-						<div class="card-body bg-base-300 font-bold">
-							<div class="pt-2">
-								<p class="mb-4">
-									There is no password recovery available in Swarm City.
-								</p>
+	if (showPrompt)
+		return (
+			<div class="bg-gray-lt password-warning">
+				<div class="close">
+					<UserCreateStop />
+				</div>
+				<div class="container">
+					<main class="flex-space">
+						<header>
+							<h1>Choose a password.</h1>
+						</header>
+						<div class="warning-box">
+							<img src={warningBlue} />
+							<div>
+								<p>There is no password recovery available in Swarm City.</p>
 								<p>Choose your password with care.</p>
 							</div>
 						</div>
-					</div>
+						<div class="btns">
+							<ButtonRoundArrow onClick={() => setShowPrompt(false)} />
+						</div>
+					</main>
 				</div>
-			) : (
-				<>
-					<div class="form-control mb-4">
+			</div>
+		)
+	return (
+		<div class="bg-gray-lt choose-password">
+			<div class="close">
+				<UserCreateStop />
+			</div>
+			<div class="container">
+				<main class="flex-space">
+					<header>
+						<h1>Choose a password.</h1>
+					</header>
+					<form>
 						<input
 							type="password"
-							class={cn(
-								'input',
-								'input-bordered',
-								passwordsFilledIn && !passwordValid && 'input-error'
-							)}
-							placeholder="Password"
-							onChange={(event) => changePassword(0, event)}
+							id="password"
+							placeholder="password"
+							onChange={(e) => setPassword(e.currentTarget.value)}
 						/>
-					</div>
-					<div class="form-control mb-4">
 						<input
 							type="password"
-							class={cn(
-								'input',
-								'input-bordered',
-								passwordsFilledIn && !passwordValid && 'input-error'
-							)}
-							placeholder="Confirm password"
-							onChange={(event) => changePassword(1, event)}
+							id="passwordConfirm"
+							placeholder="confirm password"
+							onChange={(e) => setPassword2(e.currentTarget.value)}
+						/>
+					</form>
+					<p class="error">{password !== password2 && 'Password mismatch'}</p>
```

Comment:
I feel like we lost a feature here, as displaying "Password mismatch" when the user didn't even get a chance to input the confirmation seems like bad UX. Also this introduces a layout shift, which didn't happen before.

Reply:
Good point and I think the UX should be different. IMHO when user clicks on the avatar without having any, they should right away be asked to choose a file or take picture. Then we can always have cropper with something in it and there is one less click needed from the user.

Reply:
Should be fixed now